contracts/takasure/TakasurePool.sol

```
@@ -349,51 +357,94 @@ contract TakasurePool is
+        _cancelMembership(memberWallet);
+    }
+
     function recurringPayment() external {
-        if (reserve.members[msg.sender].memberState != MemberState.Active) {
+        // The member can pay the contribution if active or defaulted but at least 30 days have passed from the new year
+        Member memory member = reserve.members[msg.sender];
+        if (
+            member.memberState != MemberState.Active && member.memberState != MemberState.Defaulted
+        ) {
             revert TakasureErrors.TakasurePool__WrongMemberState();
         }
         uint256 currentTimestamp = block.timestamp;
-        uint256 membershipStartTime = reserve.members[msg.sender].membershipStartTime;
-        uint256 membershipDuration = reserve.members[msg.sender].membershipDuration;
-        uint256 lastPaidYearStartDate = reserve.members[msg.sender].lastPaidYearStartDate;
+        uint256 membershipStartTime = member.membershipStartTime;
+        uint256 membershipDuration = member.membershipDuration;
+        uint256 lastPaidYearStartDate = member.lastPaidYearStartDate;
         uint256 year = 365 days;
         uint256 gracePeriod = 30 days;
+        uint256 firstLimitTimestamp = membershipStartTime + membershipDuration; // The complete membership duration
+        uint256 secondLimitTimestamp = lastPaidYearStartDate + year + gracePeriod;
 
-        if (
-            currentTimestamp > lastPaidYearStartDate + year + gracePeriod ||
-            currentTimestamp > membershipStartTime + membershipDuration
-        ) {
+        if (currentTimestamp > firstLimitTimestamp) {
             revert TakasureErrors.TakasurePool__InvalidDate();
+        } else if (currentTimestamp >= secondLimitTimestamp) {
+            // Update the state, this will allow to cancel the membership
+            reserve.members[msg.sender].memberState = MemberState.Defaulted;
+
+            emit TakasureEvents.OnMemberDefaulted(reserve.members[msg.sender].memberId, msg.sender);
```

Hey Maikel, i can see that if a user tries to make a payment after his grace days passing, then his state is updated to defaulted. This behavior is out of the concern of this function. Meaning, `recurringPayment ` must only focus on making the payment, and failing if its the wrong user state. Thus here you can simply return an error such as `paymentTooLate`. It will be the responsibility of the CS to mark the user as defaulted, or the responsibility of a cron job.

---

Commit a4d5b2c